jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/util/SearchUtil.java

```diff
@@ -0,0 +1,159 @@
+package com.github.dart_lang.jnigen.apisummarizer.util;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.util.*;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+
+public class SearchUtil {
+  /// find files in path
+  public static Optional<List<File>> findFilesInPath(
+      String qualifiedName, String searchPath, String suffix) {
+    var s = qualifiedName.replace(".", "/");
+    var f = new File(searchPath, s + suffix);
+    if (f.exists() && f.isFile()) {
+      return Optional.of(List.of(f));
+    }
+
+    var d = new File(searchPath, s);
+    if (d.exists() && d.isDirectory()) {
+      return Optional.of(recursiveListFiles(d, file -> file.getName().endsWith(".java")));
+    }
+
+    return Optional.empty();
+  }
+
+  /// find files in jar
+  public static Optional<List<ZipEntry>> findFilesInJar(
+      String qualifiedName, JarFile jar, String suffix) {
+    String relativePath = qualifiedName.replace(".", "/");
+    var classEntry = jar.getEntry(relativePath + suffix);
+    if (classEntry != null) {
+      return Optional.of(List.of(classEntry));
+    }
+    var dirPath = relativePath.endsWith("/") ? relativePath : relativePath + "/";
```

Comment:
Platform dependant path separators. Used in different places within this file.

Reply:
It's inside JAR (used in `getEntry`). I am not sure I can use a platform dependent separator there.